__tests__/app.test.js

```diff
@@ -215,6 +215,48 @@ describe("/api/articles",() => {
             expect(body.msg).toBe("Route does not exist")
         })
     })
+
+    test("GET 200: Responds with array of article objects filtered by topic query", ()=> {
+        return request(app)
+        .get("/api/articles?topic=mitch")
+        .expect(200)
+        .then(({body}) => {
+            const {articles} = body;
+            expect(articles.length).toBe(12);
+            articles.forEach((article)=> {
+                expect(article).toEqual(
+                    expect.objectContaining({
+                        article_id: expect.any(Number),
+                        title: expect.any(String),
+                        topic: 'mitch',
+                        author: expect.any(String),
+                        comment_count: expect.any(Number),
+                        created_at: expect.any(String),
+                        article_img_url: expect.any(String),
+                        votes: expect.any(Number),
+                    })
+                )
+            })
+        })
+    })
+
+    test("GET 400: Responds with error and msg 'Bad request!' if the topic query does not exist in the database", ()=> {
+        return request(app)
+        .get("/api/articles?topic=topic_does_not_exist_in_database")
+        .expect(400)
+        .then(({body}) => {
+            expect(body.msg).toBe("Bad request!")
+        })
+    })
```

Comment:
This instead sounds like a 404 not found to me :) 

404 - When an entry doesn't exist but _could_ theoretically exist in the database

400 - When an entry could _never_ exist in the database unless the structure was changed